### PR TITLE
docs: update content for version 13 release

### PR DESCRIPTION
**Pull request by josmar-crwdstffng**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

Content changes are based upon changes to code for version 13 release.

### CODE REVIEW

**Review by josmar-crwdstffng — COMMENTED** [review]
@jelbourn 

I reviewed and implemented your suggestions.
Please review and comment or approve.

File: aio/content/guide/using-libraries.md [hunk 1]
```diff
@@ -1,136 +1,190 @@
...
 
-3. In the component or file that uses the library, add the following code:
+1.  In the component or file that uses the library, add the following code:
 
-```
-import * as host from 'host';
-const parsedUrl = host.parse('https://angular.io');
-console.log(parsedUrl.hostname);
-```
+    <code-example format="typescript" language="typescript">
```
Comment (IgorMinar) [inline aio/content/guide/using-libraries.md:57]: this also looks suspicious. do we need to define both `format` and `language`? I haven't noticed that requirement elsewhere in our docs.

Comment (IgorMinar) [reply aio/content/guide/using-libraries.md:57]: // @petebacondarwin

File: aio/content/guide/upgrade.md [hunk 2]
```diff
@@ -1,937 +1,731 @@
 # Upgrading from AngularJS to Angular
 
-_Angular_ is the name for the Angular of today and tomorrow.<br />
-_AngularJS_ is the name for all 1.x versions of Angular.
+*Angular* is the name for the Angular of today and tomorrow.
```
Comment (IgorMinar) [inline aio/content/guide/upgrade.md:3]: oof.. this change seems to combine both formating / mechanical changes and content change. it's really hard for me to confidently say that these changes are good. I'll just have to trust the @josmar-crwdstffng here.

File: aio/content/guide/update-to-latest-version.md [hunk 3]
```diff
@@ -14,7 +14,29 @@ For information about the deprecation and removal practices of Angular, see [Ang
 
 </div>
 
-*   <!--todo: add content from ng team -->
+*   **Removal of ViewEngine**
```
Comment (IgorMinar) [inline aio/content/guide/update-to-latest-version.md:17]: could we link to RFCs, issues, or blog.angular.io posts for all of these?

**Review by IgorMinar — COMMENTED** [review]
Overall this looks really good. I found a few small things that I'd appreciate tweaking. Can you please take a look @josmar-crwdstffng ? thank you!

**Comment by IgorMinar** [comment]
one more thing - if we are not going to have more docs (e.g. migration guides) in v13 (I don't know if any where planned, I don't think so...), then we could consider turning the "Update to v13" drop down to a leaf menu item (and remove the "Overview" menu sub item.

Screen Shot 2021-10-29 at 12 08 02 PM

File: aio/content/guide/using-libraries.md [hunk 4]
```diff
@@ -1,136 +1,190 @@
...
 ### Library typings
 
-Library packages often include typings in `.d.ts` files; see examples in `node_modules/@angular/material`. If your library's package does not include typings and your IDE complains, you might need to install the library's associated `@types/<lib_name>` package.
+Library packages often include typings in `.d.ts` files; see examples in `node_modules/@angular/material`.
+If the package of your library does not include typings and your IDE complains, you might need to install the `@types/<lib_name>` package with the library.
 
 For example, suppose you have a library named `d3`:
 
-<code-example language="bash">
+<code-example format="shell" language="shell">
```
[the remark this answers is not in the document]
Comment (josmar-crwdstffng) [reply aio/content/guide/using-libraries.md:24]: @IgorMinar @petebacondarwin 

I replaced the use of `bash` with `shell`, since the 2 are used interchangeably in the docs.
I am working to get all of the inline code removed from the docs to help reduce this type of confusion.
Feel free to counter my suggestion, since I am happy to move in the same direction as long as we are all aligned.

File: aio/content/guide/using-libraries.md (hunk 1 above)
Comment (josmar-crwdstffng) [reply aio/content/guide/using-libraries.md:57]: @IgorMinar @petebacondarwin

I added the format, since the format and language are used inconsistently in the docs.

Per my other comment.
```md
I am working to get all of the inline code removed from the docs to help reduce this type of confusion.
Feel free to counter my suggestion, since I am happy to move in the same direction as long as we are all aligned.
```

File: aio/content/guide/using-libraries.md [hunk 5]
```diff
@@ -1,136 +1,190 @@
-# Using published libraries
+# Use published libraries
```
[the remark this answers is not in the document]
Comment (josmar-crwdstffng) [reply aio/content/guide/using-libraries.md:1]: @IgorMinar 

As a page title the "Using" infers that you are already in the process of doing the action "use".
As you have iterated (and I inferred), the purpose of the topic is to describe the usage of the published library.

Suggestion

```md
Usage of Angular libraries published to npm
```

File: aio/content/guide/using-libraries.md [hunk 6]
```diff
@@ -1,136 +1,190 @@
...
-To integrate reusable library code into an application, you need to install the package and import the provided functionality where you will use it. For most published Angular libraries, use the Angular CLI `ng add <lib_name>` command.
+Libraries are published as [npm packages][AioGuideNpmPackages], usually together with schematics that integrate them with the Angular CLI.
+To integrate reusable library code into an application, you need to install the package and import the provided functionality where you will use it.
+For most published Angular libraries, use the Angular CLI `ng add <lib_name>` command.
 
-The `ng add` command uses a package manager such as [npm](https://www.npmjs.com/) or [yarn](https://yarnpkg.com/) to install the library package, and invokes schematics that are included in the package to other scaffolding within the project code, such as adding import statements, fonts, and themes.
+The `ng add` command uses a package manager such as [npm][NpmjsMain] or [yarn][YarnpkgMain] to install the library package, and invokes schematics that are included in the package to other scaffolding within the project code, such as adding import statements, fonts, and themes.
```
[the remark this answers is not in the document]
Comment (josmar-crwdstffng) [reply aio/content/guide/using-libraries.md:12]: @ igorminar

Great question.
This is a short list of the "why".

*   Reference links reduce the length of inline links.
*   Reference links reduce mistakes made in repeated URIs by placing the URI and alt text is a single location.
*   Reference links increase search-friendliness in the markdown (or any code that supports whole document reference links).
*   Reference links help you identify improperly escaped parenthesis in the markdown text.

tldr;  

You end up with more consistent and tighter markdown that is easier to maintain.

File: aio/content/guide/using-libraries.md [hunk 7]
```diff
@@ -1,136 +1,190 @@
...
 
 A published library typically provides a README or other documentation on how to add that lib to your application.
-For an example, see the [Angular Material](https://material.angular.io/) documentation.
+For an example, see the [Angular Material][AngularMaterialMain] documentation.
 
 ### Library typings
 
-Library packages often include typings in `.d.ts` files; see examples in `node_modules/@angular/material`. If your library's package does not include typings and your IDE complains, you might need to install the library's associated `@types/<lib_name>` package.
+Library packages often include typings in `.d.ts` files; see examples in `node_modules/@angular/material`.
```
[the remark this answers is not in the document]
Comment (josmar-crwdstffng) [reply aio/content/guide/using-libraries.md]: Updated.

```md
Typicaly, library packages include typings in `.d.ts` files; see examples in `node_modules/@angular/material`.
```

File: aio/content/guide/update-to-latest-version.md (hunk 3 above)
Comment (josmar-crwdstffng) [reply aio/content/guide/update-to-latest-version.md:17]: @IgorMinar 
Can you share the URIs that you are referencing?

Updated.

```md
*   **Disk Cache**

    Enables the persistent build cache by default for all applications.
    See [PR&nbsp;#21545][GithubAngularAngularCliIssues21545].
```

**Review by josmar-crwdstffng — COMMENTED** [review]
@IgorMinar 
I reviewed and either commented or implemented your suggestions.

Please review and comment or approve.

File: aio/content/guide/using-libraries.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (IgorMinar) [reply aio/content/guide/using-libraries.md:12]: Thanks for the explanation, Joe. I personally prefer inlined urls if the urls are simple and not repeated, but I'll honor this usage within angular.io for consistency sake.

File: aio/content/guide/using-libraries.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (IgorMinar) [reply aio/content/guide/using-libraries.md:24]: if they are both functionally the same then I'm fine with either, as long as we use them consistently. :-)

File: aio/content/guide/using-libraries.md (hunk 1 above)
Comment (IgorMinar) [reply aio/content/guide/using-libraries.md:57]: but do we actually need both? does the rendering of the snippet change if they are both specified as opposed to specifying only one of them (for example just the `language`)?

File: aio/content/guide/update-to-latest-version.md (hunk 3 above)
Comment (IgorMinar) [reply aio/content/guide/update-to-latest-version.md]: sure. I'll post in separate comments.

**Comment by josmar-crwdstffng** [comment]
@IgorMinar 

> one more thing - if we are not going to have more docs (e.g. migration guides) in v13 (I don't know if any where planned, I don't think so...), then we could consider turning the "Update to v13" drop down to a leaf menu item (and remove the "Overview" menu sub item.
> 
> Screen Shot 2021-10-29 at 12 08 02 PM

Updated to be a "leaf" topic instead of a drop-down with overview topic.

File: aio/content/guide/update-to-latest-version.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+    Removes all support for Microsoft Internet Explorer 11 \(IE11\).
+
+*   **Testbed module teardown**
+
+    Adds the option in `initTestEnvironment` to completely remove test environments from an application.
+
+*   **Factory-less `Component` and `NgModule` APIs**
+
+    Removes the need for the `ComponentFactoryResolver` API.
```
Comment (IgorMinar) [inline aio/content/guide/update-to-latest-version.md:35]:
```suggestion
```

There was a last minute change and we ended up not deprecating this API. // cc: @AndrewKushnir

File: aio/content/guide/update-to-latest-version.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+*   **`$localize` tagged message strings**
+
+    Adds documentation for the Angular `$localize` API and tagged message strings.
```
Comment (IgorMinar) [inline aio/content/guide/update-to-latest-version.md:39]:
```suggestion
```

This was not changed or deprecated. It's a feature / new docs, so I don't believe that it belongs here.

File: aio/content/guide/using-libraries.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (aikidave) [reply aio/content/guide/using-libraries.md:12]: Joe's list is correct. The format has a lot of great benefits, even though they're not quite as "inferrable" as inline URLs.

**Review by aikithoughts — APPROVED** [review]
Reviewed-for: global-docs-approvers

Just need engineering to sign off!

**Review by IgorMinar — APPROVED** [review]
How about we merge this as is and address the pending comments in a follow up PR.

**Review by alxhub — APPROVED** [review]

**Comment by alxhub** [comment]
Per @IgorMinar, we'll merge this now and address any minor feedback in a followup PR.

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 3d1ac6a0097568c99db5deb32b673ced48e9f785.

File: aio/content/guide/update-to-latest-version.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+| [PR&nbsp;#43642][GithubAngularAngularPull43642] | TypeScript versions older than `4.4.2` are no longer supported.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```
[the remark this answers is not in the document]
Comment (josmar-crwdstffng) [reply aio/content/guide/update-to-latest-version.md:59]: @alxhub

The list is ordered by PR number from oldest to latest.
The details do not provide any ordination.
Should we abandon the sequential list in favor of a flowing faux headings found under the **Changes and deprecations in version 13** heading?

**Review by josmar-crwdstffng — COMMENTED** [review]
@IgorMinar 
@petebacondarwin 
@alxhub
I reviewed and implemented the outstanding changes in #44009 .
Please review and either comment or approve.
